source/Calamari/Commands/Java/JavaLibraryCommand.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Calamari.Commands.Support;
+using Calamari.Deployment;
+using Calamari.Deployment.Conventions;
+using Calamari.Deployment.Features.Java;
+using Calamari.Integration.EmbeddedResources;
+using Calamari.Integration.FileSystem;
+using Calamari.Integration.Processes;
+using Calamari.Integration.Scripting;
+using Calamari.Integration.ServiceMessages;
+
+namespace Calamari.Commands.Java
+{
+    [Command("java-step", Description = "Invokes the Octopus java library")]
+    public class JavaLibraryCommand : Command
```

Comment:
All java commands will go through this one step. They all need variables and the JarTool proxy class, they only differ on what convention is ran.